### ISSUE TITLE
ViewModel RaisePropertyChanged patterns

### ISSUE THREAD

**Issue by Noggog** [wabbajack-tools/wabbajack#75]
Not really a feature request, but more of a pattern suggestion to make the code more robust and succinct that you guys might find useful.

I noticed that the RaisePropertyChanged calls typically look like this:
```
public string Mode
{
    get => _mode;
    set
    {
        _mode = value;
        OnPropertyChanged("Mode");
    }
}
```

I've got two small suggestions to consider:
**1)  nameof() concepts for when strings need to match a member/function name**
So something like
```
public string Mode
{
    get => _mode;
    set
    {
        _mode = value;
        OnPropertyChanged(nameof(Mode));
    }
}
```
This is a bit more robust, as now if anyone uses a refactoring tool to rename Mode, the string passed to OnPropertyChanged is powered by the compiler now, and implicitly changed by the refactor.  Helps avoid future gotcha magic string bugs.

**2)  RaiseAndSetIfChanged function might be useful to utilize:**
```
protected void RaiseAndSetIfChanged<T>(
    ref T item,
    T newItem,
    [CallerMemberName] string propertyName = null)
{
    if (EqualityComparer<T>.Default.Equals(item, newItem)) return;
    item = newItem;
    this.RaisePropertyChanged(propertyName);
}
```

Might put it in some common base class like `ViewModel`?   Either way, would be used to simplify the above code to something like this:
```
public string Mode
{
    get => _mode;
    set => RaiseAndSetIfChanged(ref _mode, value);
}
```
Makes my first suggestion kind of moot, haha.

I definitely love refactoring, so I can set up a PR with these suggestions completed.. but I felt like I should pitch the idea and see what you guys thought before spending time on it.  Especially since the changes would touch a good amount of existing files I think.  8)

**Comment by halgari** [comment]
I'm fine with a PR that refactors this

**Comment by Noggog** [comment]
One last thing for clarification on your desired setup.  I noticed the member itself `_mode` was way up in the file, nowhere close to the snippets I've been referencing above.  Is this intentional or just a vestigial byproduct of development?

Usually I prefer it something like:
```
private string _mode;
public string Mode
{
    get => _mode;
    set => RaiseAndSetIfChanged(ref _mode, value);
}
```

Usually I set up a Visual Studio snippet, so you can make a `guiprop` snippet, similar to `prop`, that would generate all the code above, letting you just type in the name/type information quickly and be done.  That works better when it's all in the same location.

**Comment by halgari** [comment]
no reason, feel free to move them.

**Comment by Noggog** [comment]
Implemented with #76
